plugins/omelasticsearch/omelasticsearch.c

```diff
@@ -247,76 +276,119 @@ ENDdbgPrintInstInfo
 
 
 /* Build basic URL part, which includes hostname and port as follows:
- * http://hostname:port/
- * Newly creates an estr for this purpose.
+ * http://hostname:port/ based on a server param
+ * Newly creates a cstr for this purpose.
  */
 static rsRetVal
-setBaseURL(instanceData *pData, es_str_t **url)
+computeBaseUrl(char* serverParam, int defaultPort, sbool useHttps, uchar **baseUrl)
 {
+#	define SCHEME_HTTPS "https://"
+#	define SCHEME_HTTP "http://"
+
 	char portBuf[64];
-	int r;
+	int r = 0;
+	char* host = serverParam;
 	DEFiRet;
 
-	*url = es_newStr(128);
-	snprintf(portBuf, sizeof(portBuf), "%d", pData->port);
-        if (pData->useHttps) {
-  		r = es_addBuf(url, "https://", sizeof("https://")-1);
+	es_str_t *urlBuf = es_newStr(256);
+	if (urlBuf == NULL) {
+		DBGPRINTF("omelasticsearch: failed to allocate es_str urlBuf in computeBaseUrl\n");
+		ABORT_FINALIZE(RS_RET_ERR);
+	}
+
+	/* Remove a trailing slash if it exists */
+	if (serverParam[strlen(serverParam)-1] == '/')
+		serverParam[strlen(serverParam)-1] = '\0';
+
+	/* Find where the hostname/ip of the server starts. If the scheme is not specified
+	  in the uri, start the buffer with a scheme corresponding to the useHttps parameter.
+	*/
+	if (strcasestr(serverParam, SCHEME_HTTP))
+		host = serverParam + strlen(SCHEME_HTTP);
+	else if (strcasestr(serverParam, SCHEME_HTTPS))
+		host = serverParam + strlen(SCHEME_HTTPS);
+	else
+		r = useHttps ?
+			es_addBuf(&urlBuf, SCHEME_HTTPS, sizeof(SCHEME_HTTPS)-1) : 
```

Comment:
Personal style? I changed the alignment, but if you prefer a different style- I am indifferent.